definitions/npm/ramda_v0.26.x/flow_v0.104.x-/test_ramda_v0.26.x_list.js

```diff
@@ -822,4 +823,40 @@ const str: string = "hello world";
     s: a,
     y: b
   }))([1, 2, 3])(["1", "2", "3"]);
+
+  describe('uniq', () => {
+    it('should accept read only array', () => {
+      const readOnlyNumbers: $ReadOnlyArray<number> = [1,1,2,3,4,3];
+      const result:$ReadOnlyArray<number> = uniq(readOnlyNumbers);
+    });
+
+    it('should fail when element type mismatches', () => {
+      const readOnlyNumbers: $ReadOnlyArray<number> = [1,1,2,3,4,3];
+      //$ExpectError
+      const result:$ReadOnlyArray<string> = uniq(readOnlyNumbers);
+
+      //$ExpectError
+      const result: Array<string> = uniq(readOnlyNumbers);
+    });
+
+    it('should accept mutable array', () => {
+      const arr: Array<string> = ['1', '2', '3'];
+      const result: Array<string> = uniq(arr);
+    });
+
+    it('should fail for a mixed element type array', () => {
+      const mix = ['1', 2, true];
+      //$ExpectError
+      const result: $ReadOnlyArray<string> = uniq(mix);
+    });
+
+    it('should fail for not read only output', () => {
+      const readOnlyNumbers: $ReadOnlyArray<number> = [1,1,2,3,4,3];
+      // $ExpectError
+      const result: Array<number> = uniq(readOnlyNumbers);
```

Comment:
This is the same case as above but with its own `it` block (generally this is a better pattern), but also this is a case we want to support rather than prevent. See earlier comment re: `$ReadOnlyArray` input and `Array` output.

Reply:
Done